### PR TITLE
GraphQL ADR: prototype resolvers for tsp-client.ts

### PR DESCRIPTION
**Pull request by marco-miller**
This PR is described in its commit message.

### CODE REVIEW

**Review by bhufmann — COMMENTED** [review]
When looking at the proposal, I understand that the proposed GraphQL server, that is implemented using Apollo server, translates GraphQL queries into TSP API REST queries towards the trace server.

My understanding of the GraphQL for our project was, that the trace server application would be the GraphQL server, serving GraphQL query requests, and on the client side (tsp-typescript-client) a GraphQL client would implemented, that connects to the GraphQL server backend (though HTTP). So, the tsp-typescript-client (and tsp-python-client respectively) would implement the client side, maybe using Apollo Client.

Or do I misunderstand something?

**Comment by marco-miller** [comment]
> When looking at the proposal, I understand that the proposed GraphQL server, that is implemented using Apollo server, translates GraphQL queries into TSP API REST queries towards the trace server.

As an initial or first proposal, yes.

> My understanding of the GraphQL for our project was, that the trace server application would be the GraphQL server, serving GraphQL query requests, and on the client side (tsp-typescript-client) a GraphQL client would implemented, that connects to the GraphQL server backend (though HTTP). So, the tsp-typescript-client (and tsp-python-client respectively) would implement the client side, maybe using Apollo Client.

Apollo Client is not necessary yet, but yes, the existing tsp-*-clients would send graph requests towards the GraphQL server, either directly or through an app using them such as this extension repo. Now, we may decide to make the GraphQL server rely directly on trace-server internals, but this proposal brings the idea of proving clients more quickly, by means of taping onto TSP REST as initial iterations.

File: doc/adr/0002/graphql-server-prototype/test [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Assumes [1] below, which assumes a running trace-server; based on [2].
+# POST is used for the curl request, but GET [3] can be used although less readable.
+# Any combinations of the supported fields can be queried, which is why GraphQL.
```
Comment (Rodrigoplp-work) [inline doc/adr/0002/graphql-server-prototype/test:4]: Did something get cut out from the end of the sentence? Was it meant to be something like "... which is why GraphQL was suggested as a solution."?

**Review by Rodrigoplp-work — CHANGES_REQUESTED** [review]

File: doc/adr/0002/graphql-server-prototype/test (hunk 1 above)
Comment (marco-miller) [reply doc/adr/0002/graphql-server-prototype/test:4]: No, not really; this way of saying does exist in English, AFAIK.

**Review by Rodrigoplp-work — APPROVED** [review]

**Pull request merged**